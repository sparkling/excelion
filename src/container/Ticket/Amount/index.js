import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import NumberFormat from 'react-number-format';

const StyledInput = styled(NumberFormat) `
  font-size: 1em;
  flex: 1;
  text-align: center
`

const Amount = ({ onValueChange, value }) =>
  (<StyledInput
      value={value}
      thousandSeparator={true}
      type="text"
      name="amount"
      onValueChange={(values) => onValueChange(values && values.floatValue)}
    />);


Amount.propTypes = {
  onValueChange: PropTypes.func.isRequired,
  value: PropTypes.number
}

export default Amount;
