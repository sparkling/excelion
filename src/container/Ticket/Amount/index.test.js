import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import Amount from './index.js';

const renderComponent = (props = {}) => shallow(<Amount {...props} />);
const defaultRender = renderComponent({onValueChange:()=>{}, value:123}).children().dive();

describe('<Amount />', () => {
  it('should render the same snapshot', () => {
    const tree = renderer.create(<Amount onValueChange={()=>{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the NumberFormat component', () => {
    expect(defaultRender.find('NumberFormat').length).toBe(1);
  });

  it('should use a thousand separator', () => {
    expect(defaultRender.prop('thousandSeparator')).toBe(true);
  });

  it('should call onChange when selection is made', () => {
    const onValueChange = sinon.spy();
    renderComponent({ onValueChange:onValueChange }).children().dive().simulate('valueChange');
    expect(onValueChange.called).toBe(true);
  });
});
