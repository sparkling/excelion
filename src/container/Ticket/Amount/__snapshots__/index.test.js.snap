// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Amount /> should render the same snapshot 1`] = `
.c0 {
  width: 10em;
  font-size: 1em;
  padding: 0.5em;
  text-align: center;
}

<form>
  <input
    className="c0"
    name="amount"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    type="text"
    value=""
  />
</form>
`;
